FirebaseMessaging/Apps/AdvancedSample/Podfile

```diff
@@ -3,15 +3,35 @@ use_frameworks!
 source 'https://github.com/firebase/SpecsStaging.git'
 source 'https://cdn.cocoapods.org/'
 
-target 'AdvancedSample' do
-  platform :ios, '13.0'
-
+def shared_pods
   pod 'FirebaseCore', :path => '../../../'
   pod 'GoogleUtilities', :path => '../../../'
   pod 'FirebaseMessaging', :path => '../../../'
   pod 'FirebaseInstanceID', :path => '../../../'
   pod 'FirebaseCoreDiagnostics', :path => '../../../'
   pod 'GoogleDataTransport', :path => '../../../'
   pod 'FirebaseInstallations', :path => '../../../'
+end
+
+target 'AdvancedSample' do
+  platform :ios, '13.0'
+  shared_pods
+end
 
+target 'NotificationServiceExtension' do
+  platform :ios, '13.0'
+  shared_pods
+end
+
+target 'AppClips' do
+  use_frameworks! :linkage => :static
```

Comment:
It's kind of weird to put the linkage here versus near the top of the file. I don't think different targets can be linked differently.

Reply:
I got an crash error "dyld: Library not loaded. Image not found" for Promise library linking when running app clips and  I put this in podfile that fixes the issue.

Reply:
Do you get the dyld error when `use_frameworks! :linkage => :static` is at the top of the file and when it's not there at all?

Reply:
this is only for app clip that is why I put it inside app clip target, originally I had use_framework! on top of file that crashed app clips

Reply:
Hmm, I would think `use_frameworks! :linkage => :static` at the top should be equivalent?

Reply:
So I guess we can apply this to all targets, I just thought we have to make special case for static linkage for app clips.

Reply:
Unless there's something I don't understand about App Clips, it's not possible to link differently in the same app.

Reply:
mmmm, I put it on top and seems working for all targets. But I don't know why app clips needs this.